### PR TITLE
Convert DateTime serialization in UTC

### PR DESCRIPTION
**Pull request by QilongTang**
Please Note:
1. Before submitting the PR, please review [How to Contribute to Dynamo](https://github.com/DynamoDS/Dynamo/blob/master/CONTRIBUTING.md)
2. Dynamo Team will meet 1x a month to review PRs found on Github (Issues will be handled separately)
3. PRs will be reviewed from oldest to newest
4. If a reviewed PR requires changes by the owner, the owner of the PR has 30 days to respond. If the PR has seen no activity by the next session, it will be either closed by the team or depending on its utility will be taken over by someone on the team
5. PRs should use either Dynamo's default PR template or [one of these other template options](https://github.com/DynamoDS/Dynamo/wiki/Choosing-a-Pull-Request-Template) in order to be considered for review.
6. PRs that do not have one of the Dynamo PR templates completely filled out with all declarations satisfied will not be reviewed by the Dynamo team.
7. PRs made to the `DynamoRevit` repo will need to be cherry-picked into all the DynamoRevit Release branches that Dynamo supports. Contributors will be responsible for cherry-picking their reviewed commits to the other branches after a `LGTM` label is added to the PR.

### Purpose

[QNTM-2572](https://jira.autodesk.com/browse/QNTM-2572) DateTime serializes incorrectly into inputs block Dynamo

This PR:
1. Changing datetime node serialization and its corresponding input serialization in utc form referring to https://www.w3.org/TR/NOTE-datetime

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

@ColinDayOrg 

### FYIs

@mjkkirschner  @alfarok @ramramps

### CODE REVIEW

File: src/Libraries/CoreNodeModels/Input/DateTime.cs [hunk 1]
```diff
@@ -41,7 +41,7 @@ public override NodeInputData InputData
                     Type = NodeInputData.getNodeInputTypeFromType(typeof(System.DateTime)),
                     Description = this.Description,
                     //format dateTime with swagger spec in mind:  ISO 8601.
-                    Value = Value.ToString("s",CultureInfo.InvariantCulture),
+                    Value = Value.ToUniversalTime().ToString("yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fff'Z'"),
```
Comment (ramramps) [inline src/Libraries/CoreNodeModels/Input/DateTime.cs]: did you check AVP format?

Comment (gregmarr) [reply src/Libraries/CoreNodeModels/Input/DateTime.cs]: Is this better than using the function built specifically for this?

    Value = XmlConvert.ToString(Value, XmlDateTimeSerializationMode.Utc);

**Comment by ColinDayOrg** [comment]
Other than addressing Greg's comment, LGTM.

File: src/Libraries/CoreNodeModels/Input/DateTime.cs (hunk 1 above)
Comment (QilongTang) [reply src/Libraries/CoreNodeModels/Input/DateTime.cs]: @ramramps I did check that AVP is using UTC, so the round trip should be good. @gregmarr @ColinDayOrg  I was a bit hesitate to use XML serialization code but I just found there is the easiest way using `ToString("O")`
```
The "O" or "o" standard format specifier (and the "yyyy'-'MM'-'dd'T'HH':'mm':'ss'.'fffffffK" custom format string) takes advantage of the three ways that ISO 8601 represents time zone information to preserve the Kind property of DateTime values:
The time zone component of DateTimeKind.Utc date and time values uses "Z" (which stands for zero offset) to represent UTC.
```

**Comment by QilongTang** [comment]
The CI error does not seem related to my change. Merging

**Pull request merged**